webapp/packages/plugin-resource-manager/src/NavResourceNodeService.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { NavNodeInfoResource, NavTreeResource } from '@cloudbeaver/core-app';
+import { injectable } from '@cloudbeaver/core-di';
+
+import { ResourceManagerResource } from './ResourceManagerResource';
+import { RESOURCES_NODE_PATH } from './RESOURCES_NODE_PATH';
+
+interface IResourceData {
+  projectId: string;
+  resourcePath: string;
+}
+
+export const PROJECT_NODE_TYPE = 'rm.project';
+export const RESOURCE_NODE_TYPE = 'rm.resource';
+
+@injectable()
+export class NavResourceNodeService {
+  constructor(
+    private readonly navTreeResource: NavTreeResource,
+    private readonly navNodeInfoResource: NavNodeInfoResource,
+    private readonly resourceManagerResource: ResourceManagerResource,
+  ) { }
+
+  async saveScript(projectId: string, name: string, script: string, path?: string) {
+    const resourcePath = `${path ? path + '/' : ''}${name}`;
+    await this.resourceManagerResource.createResource(projectId, resourcePath, false);
+    await this.resourceManagerResource.writeResource(projectId, resourcePath, script);
+    await this.navTreeResource.refreshTree(`${RESOURCES_NODE_PATH}/${projectId}${path ? '/' + path : ''}`);
+  }
```

Comment:
`async saveScript(folderNodeId: string, name: string, script: string)`
`const resourceData = this.getResourceData(folderNodeId);`
`...`
`await this.navTreeResource.refreshTree(folderNodeId);`